### PR TITLE
Use service instead of node address in consul node response

### PR DESCRIPTION
**Pull request by jabnouneo**
Update DiscoveryHiveCluster to cope with the latest mesos network isolation changes  (https://confluence.criteois.com/display/MESOS/Network+isolation+on+Mesos)

### CODE REVIEW

**Review by DamienLesage — APPROVED** [review]

File: presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/DiscoveryHiveCluster.java [hunk 1]
```diff
@@ -148,6 +148,9 @@ public HiveMetastoreClient createMetastoreClient()
         List<ServiceHealth> response = result.getResponse();
         return response.stream().map(uri -> {
             String host = uri.getNode().getNode();
+            if (host == null || host.isEmpty()) {
```
Comment (aroville) [inline presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/DiscoveryHiveCluster.java]: The logic should be to fallback to node address when service address is absent:
host = service.address if (service.address != null) else node.address

Comment (mliroz) [reply presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/DiscoveryHiveCluster.java]: So I guess in this case the implementation should look like:
```
String host = uri.getService().getAddress();
if (host == null || host.isEmpty()) {
  host = uri.getNode().getAddress();
}
```

**Review by aroville — APPROVED** [review]

**Review by mliroz — APPROVED** [review]
LGTM

**Comment by rclaude** [comment]
This change needs to be made on our current repository & current branch: https://github.com/criteo-forks/presto/commits/344-criteo. 
Documentation on how we manage presto code changes: https://confluence.criteois.com/display/PRESTO/Presto+Version+Bump
I'll archive this repository to protect us from repeating this confusion.

**Pull request closed without merging**